packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -0,0 +1,208 @@
+ * Until we refactor it, just copying the code is the simplest solution.
+ *
+ * @param {string} search
+ * @param {Object} [searchArguments]
+ * @param {number} [searchArguments.isInitialSuggestions]
+ * @param {number} [searchArguments.type]
+ * @param {number} [searchArguments.subtype]
+ * @param {number} [searchArguments.page]
+ * @param {Object} [editorSettings]
+ * @param {boolean} [editorSettings.disablePostFormats=false]
+ * @return {Promise<Object[]>} List of suggestions
+ */
+
+const fetchLinkSuggestions = async (
+	search,
+	{ isInitialSuggestions, type, subtype, page, perPage: perPageArg } = {},
+	{ disablePostFormats = false } = {}
+) => {
+	const perPage = perPageArg || isInitialSuggestions ? 3 : 20;
+
+	const queries = [];
+
+	if ( ! type || type === 'post' ) {
+		queries.push(
+			apiFetch( {
+				path: addQueryArgs( '/wp/v2/search', {
+					search,
+					page,
+					per_page: perPage,
+					type: 'post',
+					subtype,
+				} ),
+			} ).catch( () => [] ) // fail by returning no results
+		);
+	}
+
+	if ( ! type || type === 'term' ) {
+		queries.push(
+			apiFetch( {
+				path: addQueryArgs( '/wp/v2/search', {
+					search,
+					page,
+					per_page: perPage,
+					type: 'term',
+					subtype,
+				} ),
+			} ).catch( () => [] )
+		);
+	}
+
+	if ( ! disablePostFormats && ( ! type || type === 'post-format' ) ) {
+		queries.push(
+			apiFetch( {
+				path: addQueryArgs( '/wp/v2/search', {
+					search,
+					page,
+					per_page: perPage,
+					type: 'post-format',
+					subtype,
+				} ),
+			} ).catch( () => [] )
+		);
+	}
+
+	return Promise.all( queries ).then( ( results ) => {
+		return map(
+			flatten( results )
+				.filter( ( result ) => !! result.id )
+				.slice( 0, perPage ),
+			( result ) => ( {
+				id: result.id,
+				url: result.url,
+				title: decodeEntities( result.title ) || __( '(no title)' ),
+				type: result.subtype || result.type,
+			} )
+		);
+	} );
+};
+
+/**
+ * React hook used to compute the block editor settings to use for the post editor.
+ *
+ * @param {Object}  settings    EditorProvider settings prop.
+ * @param {boolean} hasTemplate Whether template mode is enabled.
+ *
+ * @return {Object} Block Editor Settings.
+ */
+function useBlockEditorSettings( settings, hasTemplate ) {
+	const {
+		reusableBlocks,
+		hasUploadPermissions,
+		canUseUnfilteredHTML,
+		isTitleSelected,
+	} = useSelect( ( select ) => {
+		const { canUserUseUnfilteredHTML, isPostTitleSelected } = select(
+			editorStore
+		);
+		const { canUser } = select( coreStore );
+
+		return {
+			canUseUnfilteredHTML: canUserUseUnfilteredHTML(),
+			reusableBlocks: select( 'core' ).getEntityRecords(
+				'postType',
+				'wp_block',
+				{ per_page: -1 }
+			),
+			hasUploadPermissions: defaultTo(
+				canUser( 'create', 'media' ),
+				true
+			),
+			// This selector is only defined on mobile.
+			isTitleSelected: isPostTitleSelected && isPostTitleSelected(),
+		};
+	} );
```

Comment:
This `useSelect` call is missing a dependency array.

Reply:
Thanks 🙏